src/power_grid_model/_utils.py

```diff
@@ -749,3 +768,18 @@ def get_dataset_type(data: Dataset) -> DatasetType:
         raise ValueError("The dataset type could not be deduced because multiple dataset types match the data.")
 
     return next(iter(candidates))
+
+
+def get_comp_batch_size(comp_data: dict) -> int:
```

Comment:
Essentially dict

Reply:
Yes, but if it changes, having the specific data type hinted here may help us in the future. It also improves readability because it adds context to the dictionary.

Reply:
See below.